cpp/src/parquet/encoding.h

```diff
@@ -356,7 +425,8 @@ std::unique_ptr<typename EncodingTraits<DType>::Decoder> MakeTypedDecoder(
     Encoding::type encoding, const ColumnDescriptor* descr = NULLPTR) {
   using OutType = typename EncodingTraits<DType>::Decoder;
   std::unique_ptr<Decoder> base = MakeDecoder(DType::type_num, encoding, descr);
-  return std::unique_ptr<OutType>(dynamic_cast<OutType*>(base.release()));
+  auto out = dynamic_cast<OutType*>(base.release());
+  return std::unique_ptr<OutType>(out);
```

Comment:
Curious what motivated this change. Readability?

Reply:
Actually it's debug code. `dynamic_cast` can be treacherous and at one point `out` was nullptr